src/util/newconfig/ConfigDefinition.hpp

```diff
@@ -122,7 +121,25 @@ class ClioConfigDefinition {
      * @return ValueView associated with the given key
      */
     [[nodiscard]] ValueView
-    getValue(std::string_view fullKey) const;
+    getValueView(std::string_view fullKey) const;
+
+    /**
+     * @brief Returns the specified value of given string if value exists
+     *
+     * @tparam T The type T to return
+     * @param fullKey The config key to search for
+     * @return Value of key of type T
+     */
+    template <typename T>
+    T
+    getValue(std::string_view fullKey) const
+    {
+        ASSERT(map_.contains(fullKey), "key {} does not exist in config", fullKey);
+        if (std::holds_alternative<ConfigValue>(map_.at(fullKey))) {
```

Comment:
Save `map_.at(fullKey)` into a variable to not search twice. Maybe compiler will do it for you, but it is better to be sure.